### PR TITLE
Blob mess

### PR DESCRIPTION
**Pull request by smontana30**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/BlobFormHandlerServlet.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import java.util.Arrays;
+import static java.util.Arrays.*;
+import java.util.List;
+ 
+@WebServlet("/form-blob")
+public class BlobFormHandlerServlet extends HttpServlet {
+ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
```
Comment (NicholasCobbFromGoogle) [inline portfolio/src/main/java/com/google/sps/servlets/BlobFormHandlerServlet.java]: nit: remove this empty line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -96,6 +94,6 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       
       datastore.put(taskEntity);
 
-      response.sendRedirect("/data");
+       response.sendRedirect("/index.html");
```
Comment (NicholasCobbFromGoogle) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: remove a space at the beginning of this line

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -29,15 +36,58 @@ function loadComments() {
...
+function fetchBlobstoreUrlAndShowForm() {
+  fetch('/blobstore-upload')
+      .then((response) => response.text())
+      .then((imageUploadUrl) => {
+        const messageForm = document.getElementById('my-form');
+        messageForm.action = imageUploadUrl;
+      });
+}
+ 
+
```
Comment (NicholasCobbFromGoogle) [inline portfolio/src/main/webapp/script.js:48]: nit: remove empty line

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -29,15 +36,58 @@ function loadComments() {
         });
     });
 }
-
+ 
+
```
Comment (NicholasCobbFromGoogle) [inline portfolio/src/main/webapp/script.js]: nit: remove empty line

File: portfolio/src/main/webapp/uploadscript.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+function fetchBlobstoreUrlAndShowForm() {
+  fetch('/blobstore-upload')
+      .then((response) => {
+        return response.text();
+      })
+      .then((imageUploadUrl) => {
+        const messageForm = document.getElementById('my-form');
+        messageForm.action = imageUploadUrl;
+        messageForm.classList.remove('hidden');
+      });
```
Comment (NicholasCobbFromGoogle) [inline portfolio/src/main/webapp/uploadscript.js]: Some of the indentation appears to be 2 spaces per indentation level mixed with 4 spaces per indentation level, please choose one or the other but do not mix and match

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -75,10 +125,11 @@ function deleteAll() {
         comments[i].remove();
     }
 }
-
+ 
+
```
Comment (NicholasCobbFromGoogle) [inline portfolio/src/main/webapp/script.js]: nit: remove extra empty line

File: portfolio/src/main/webapp/login.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
```
Comment (mbehrlich) [inline portfolio/src/main/webapp/login.html:1]: I think this file can be deleted now.

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -29,15 +36,58 @@ function loadComments() {
...
+  trexInfoWindow.open(map, trexMarker);
+}
+ 
+ 
 function createListElement(comment) {
     const commentElement = document.createElement('li');
     commentElement.className = 'comment';
     commentElement.id = comment.id;
-
+
```
Comment (mbehrlich) [inline portfolio/src/main/webapp/script.js:84]: Nit: I think there are a lot of these little whitespaces that have been accidentally added. If possible, they should be removed (but NOT the new line, just the extra spaces).

File: portfolio/src/main/webapp/uploadBlob.html [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    <link rel="stylesheet" href="blobstyle.css">
+  </head>
+  <body onload="fetchBlobstoreUrlAndShowForm()">
+    <h1>Blobstore Fetch Example</h1>
+    <p>Type a message and click submit:</p>
+
+    <form id="my-form" class="hidden" method="POST" enctype="multipart/form-data" >
+      <p>Type some text:</p>
+      <textarea name="message"></textarea>
+      <br/>
```
Comment (mbehrlich) [inline portfolio/src/main/webapp/uploadBlob.html]: Remove line break elements and use css to make margins here and below.

**Review by mbehrlich — APPROVED** [review]

**Review by NicholasCobbFromGoogle — APPROVED** [review]

**Pull request merged**